signalfx/__init__.py

```diff
@@ -156,27 +158,40 @@ def send_event(self, event_type, dimensions=None, properties=None):
             self._api_endpoint, self._API_ENDPOINT_SUFFIX),
             session=self._api_session,)
 
-    def _send_thread(self):
+    def _start_thread(self):
+        # Locking the variable tha tracks the thread status
+        # 'self._thread_running' to make it an atomic operation.
+        _lock = Lock()
```

Comment:
Almost wanted to make this a class variable self._lock so we don't initialize every time, buts its only used here. Thoughts?